docs/notebooks/doc2vecc_newsgroup20.ipynb

```diff
@@ -0,0 +1,372 @@
+   ],
+   "source": [
+    "df_positive_negative.head(5)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 133,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def cosine_similarity(a, b):\n",
+    "    \"\"\"identical vectors should yield similarity of 1.0\"\"\"\n",
+    "    return np.dot(a, b) / (np.linalg.norm(a) * np.linalg.norm(b))\n",
+    "\n",
+    "\n",
+    "def sigmoid(x):\n",
+    "    return 1 / (1 + np.exp(-x))\n",
+    "\n",
+    "\n",
+    "def compare_vectors(docvecs_list, doc_x, doc_y, doc_z):\n",
+    "    \"\"\"Returns True if doc_y is closer to doc_x than doc_z\"\"\"  \n",
+    "    return cosine_similarity(docvecs_list[doc_x], docvecs_list[doc_y]) \\\n",
+    "           > cosine_similarity(docvecs_list[doc_x], docvecs_list[doc_z])\n",
+    "\n",
+    "\n",
+    "def compare_distances(docvecs_list, doc_x, doc_y, doc_z):\n",
+    "    \"\"\" returns 1.0 if x and y are identical, while x and z are very not indentical\"\"\"\n",
+    "    return 1.0 - sigmoid(\n",
+    "        np.linalg.norm(docvecs_list[doc_x] - docvecs_list[doc_y]) -\n",
+    "        np.linalg.norm(docvecs_list[doc_x] - docvecs_list[doc_z]))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "doc2vecc_correctly_classified = df_positive_negative.apply(lambda x: compare_vectors(X_test_vectors_doc2vecc, *x), axis=1).sum()\n",
+    "word2vec_correctly_classified = df_positive_negative.apply(lambda x: compare_vectors(X_test_vectors_word2vec, *x), axis=1).sum()\n",
+    "\n",
+    "doc2vecc_avg_distance = df_positive_negative.apply(lambda x: compare_distances(X_test_vectors_doc2vecc, *x), axis=1).mean()\n",
+    "word2vec_avg_distance = df_positive_negative.apply(lambda x: compare_distances(X_test_vectors_word2vec, *x), axis=1).mean()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 167,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Doc2vecC with 0.2 corruption classified:\t 48222 correctly,\t 64.711882%\nWord2Vec classified:\t\t\t 52490 correctly,\t 70.439357%\n---------------------------------------------------------------------------\nDoc2vecC with 0.2 corruption average distance: \t 0.522506\nWord2Vec average distance:\t\t\t 0.527471\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"Doc2vecC with 0.2 corruption classified:\\t %i correctly,\\t %f%%\" % (doc2vecc_correctly_classified, doc2vecc_correctly_classified/len(df_positive_negative) * 100))\n",
+    "print(\"Word2Vec classified:\\t\\t\\t %i correctly,\\t %f%%\" % (word2vec_correctly_classified, word2vec_correctly_classified/len(df_positive_negative) * 100))\n",
+    "print('-'*75)\n",
+    "print(\"Doc2vecC with 0.2 corruption average distance: \\t %f\" % doc2vecc_avg_distance)\n",
+    "print(\"Word2Vec average distance:\\t\\t\\t %f\" % word2vec_avg_distance)\n"
+   ]
```

Comment:
Results so far seem to suggest this mode hurts classification accuracy. Do you suspect some other refinement or alternative measure will show an advantage of this technique?

Reply:
Was 0.2 corruption the best-performing of several values tried, or just the first one tested so far?  I think the author said the corruption rate should scale with document length, and be closer to 1 for longer documents:

https://github.com/RaRe-Technologies/gensim/issues/1159#issuecomment-297901638

Reply:
I reran the experiment and updated the ipynb. I summarized it here https://github.com/RaRe-Technologies/gensim/pull/1985#issuecomment-442482214 for visibility.